### PR TITLE
Use `modernisation-platform-terraform-environments` for environment management

### LINKED ISSUE #11: Decide on the approach of a multi-account management structure
We need to decide on the approach of a multi-account management structure.

We've previously spoken about:
- AWS Organisations and organisation accounts
- Landing Zones
- Control Tower
...and more.

We should record a decision in the ADR.

### PR DESCRIPTION
**Pull request by jakemulley**
This PR:
- Includes [ministryofjustice/modernisation-platform-terraform-environments](https://github.com/ministryofjustice/modernisation-platform-terraform-environments) for environment management
- Configures a new provider for environment management
- Retrieves secrets for environment management from AWS Secrets Manager

This PR resolves #11, and requires ministryofjustice/aws-root-account#6.

### CODE REVIEW

File: terraform/global-resources/main.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@ provider "aws" {
 
 # Environments provider
 provider "aws" {
-  alias = "environments"
+  alias  = "environments"
```
Comment (SteveMarshall) [inline terraform/global-resources/main.tf:18]: This looks like a mistake?

File: terraform/global-resources/secrets.tf [hunk 2]
```diff
@@ -32,7 +32,7 @@ data "aws_secretsmanager_secret_version" "environments_management" {
 }
 
 locals {
-  auth0_saml  = jsondecode(data.aws_secretsmanager_secret_version.auth0_saml.secret_string)
-  github_saml = jsondecode(data.aws_secretsmanager_secret_version.github_saml.secret_string)
+  auth0_saml              = jsondecode(data.aws_secretsmanager_secret_version.auth0_saml.secret_string)
+  github_saml             = jsondecode(data.aws_secretsmanager_secret_version.github_saml.secret_string)
```
Comment (SteveMarshall) [inline terraform/global-resources/secrets.tf:36]: I used to be all about lining up `=`, but now I'm not sure it's that useful… Was this you doing it, or a tool you use?

**Review by SteveMarshall — APPROVED** [review]

File: terraform/global-resources/secrets.tf (hunk 2 above)
Comment (SteveMarshall) [reply terraform/global-resources/secrets.tf:36]: Oh, these are both the GitHub action… That's a really weird formatting choice, but OK I guess? ¯\_(ツ)_/¯

Comment (jakemulley) [reply terraform/global-resources/secrets.tf:36]: This is automatically done by [ministryofjustice/codeformatter](https://github.com/ministryofjustice/github-actions/tree/main/code-formatter) :)

**Pull request merged**